pubnub/lib/src/dx/_utils/signature.dart

```diff
@@ -55,3 +55,6 @@ ${'$body' == 'null' ? '' : '$body'}''';
       .replaceAll(RegExp(r'\/'), '_')
       .replaceAll(RegExp(r'\=*$'), '');
 }
+
+String encodePathSegament(String pathSegment) =>
+    Uri.encodeFull(pathSegment).replaceAll('/', '%2F').replaceAll('#', '%23');
```

Comment:
I'm just curious, why `Uri.encodeFull` method doesn't encode the whole path segment? Looks like you have to still replace `/` with `%2F` and `#` with `%23`

Reply:
encodeFull consider `/` as path segment separation character which is correct.
We may use encodecomponent but It encodes all special characters which doesn't fit our requirement for signature calculation and we may need to reverse by selecting like this:
```dart
Uri.encodeComponent(pathSegment).replaceAllMapped(
        RegExp('%3D|%2B|%3A|%2A|%2C|%40|%24|%26|%28|%29'),
        (match) => {
              '%3D': '=',
              '%2B': '+',
              '%3A': ':',
              '%2A': '*',
              '%2C': ',',
              '%40': '@',
              '%24': '\$',
              '%26': '&',
              '%28': '(',
              '%29': ')',
            }[match.group(0)]!);
```